emf/src/main/java/io/lionweb/lioncore/java/emf/EMFMetamodelImporter.java

```diff
@@ -146,17 +146,45 @@ private void processStructuralFeatures(
         EAttribute eAttribute = (EAttribute) eFeature;
-        Property property = new Property(eFeature.getName(), classifier);
-        property.setID(ePackage.getName() + "-" + classifier.getName() + "-" + eFeature.getName());
-        property.setKey(ePackage.getName() + "-" + classifier.getName() + "-" + eFeature.getName());
-        classifier.addFeature(property);
-        property.setOptional(!eAttribute.isRequired());
-        DataType<DataType> propertyType =
-            dataTypeMapping.convertEClassifierToDataType(eFeature.getEType());
-        Objects.requireNonNull(propertyType, "Cannot convert type " + eFeature.getEType());
-        property.setType(propertyType);
-        if (eAttribute.isMany()) {
-          throw new IllegalArgumentException("EAttributes with upper bound > 1 are not supported");
+        // EAttributes with upper bound > 1 are not supported in LionWeb
+        if (!eAttribute.isMany()) {
+          Property property = new Property(eFeature.getName(), classifier);
+          property.setID(ePackage.getName() + "-" + classifier.getName() + "-" + eFeature.getName());
+          property.setKey(ePackage.getName() + "-" + classifier.getName() + "-" + eFeature.getName());
+          classifier.addFeature(property);
+          property.setOptional(!eAttribute.isRequired());
+          DataType<DataType> propertyType =
+                  dataTypeMapping.convertEClassifierToDataType(eFeature.getEType());
+          Objects.requireNonNull(propertyType, "Cannot convert type " + eFeature.getEType());
+          property.setType(propertyType);
+        }
+        // The work-around for multiple EAttributes: introduce an intermediate containment with the upper bound > 1
+        else {
+          String featureName =
+                  eFeature.getName().substring(0, 1).toUpperCase() + eFeature.getName().substring(1);
+          Concept holderConcept = new Concept( featureName + "Container");
+          holderConcept.setID(ePackage.getName() + "-" + holderConcept.getName());
+          holderConcept.setKey(ePackage.getName() + "-" + holderConcept.getName());
+          holderConcept.setAbstract(false);
+          classifier.getLanguage().addElement(holderConcept);
+
+          Property property = new Property("content", holderConcept);
+          property.setID(ePackage.getName() + "-" + holderConcept.getName() + "-" + property.getName());
+          property.setKey(ePackage.getName() + "-" + holderConcept.getName() + "-" + property.getName());
+          holderConcept.addFeature(property);
+          property.setOptional(false);
+          DataType<DataType> propertyType =
+                  dataTypeMapping.convertEClassifierToDataType(eFeature.getEType());
+          Objects.requireNonNull(propertyType, "Cannot convert type " + eFeature.getEType());
+          property.setType(propertyType);
+
+          Containment containment = new Containment(eFeature.getName(), classifier);
```

Comment:
I see multiple occurrences of creating an M2 element and then setting its ID and key in very predictable ways: could you make a helper function for this? It would help with the readability of this code.

Reply:
Shall we not use constructors for this? I see that classes like `Concept`, `Containment`, `Interface` have constructors with `key` and `id` as their parameters, but `Property`, `Enumeration` and `EnumerationLiteral` - not.
Or do you mean a helper function for constructing IDs and Keys?

Reply:
I originally meant helper functions, but it makes more sense to align the styles of constructors.